extensions/wikia/FeedsAndPosts/TopArticles.class.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Wikia\FeedsAndPosts;
+
+class TopArticles extends MediaWikiAPI {
+
+	const LIMIT = 12;
+
+	const IMAGE_WIDTH = 80;
+	const IMAGE_RATIO = 4 / 3;
+
+	public function get() {
+		$cacheTTL = 3600; // an hour
+
+		return \WikiaDataAccess::cache( wfMemcKey( 'feeds-top-articles' ), $cacheTTL, function () {
+			$responseData = [];
+			$mainPageId = \Title::newMainPage()->getArticleID();
+			$insightData =
+				( new \InsightsContext( new \InsightsPopularPagesModel() ) )->fetchData();
```

Comment:
did you consider using this api to fetch articles ordered by pageviews? https://github.com/Wikia/app/blob/9b8b680f9bbf3998dd9f02832b5aaf865998ebc5/includes/wikia/api/ArticlesApiController.class.php#L102
it allows to filter by namespaces

Reply:
Changed to ```$data = \F::app()->sendRequest( 'ArticlesApi', 'getTop', $params )->getData();```